homeassistant/components/opengarage/cover.py

```diff
@@ -130,16 +130,16 @@ async def async_open_cover(self, **kwargs):
         self._state = STATE_OPENING
         await self._push_button()
 
-    async def async_update(self):
-        """Get updated status from API."""
-        status = await self._open_garage.update_state()
+    @callback
+    def _update_attr(self) -> None:
+        """Update the state and attributes."""
+        status = self.coordinator.data
         if status is None:
             _LOGGER.error("Unable to connect to OpenGarage device")
             self._attr_available = False
             return
 
-        if self.name is None and status["name"] is not None:
-            self._attr_name = status["name"]
+        self._device_name = self._attr_name = status["name"]
```

Comment:
Can we define the `self._attr_name` under __init__ or does the name value changes during update.

Reply:
It might change during update